Runtime/C#/Runtime/BebopWriter.cs

```diff
@@ -123,9 +144,16 @@ private void GrowBy(int amount)
             }
             if (Length + amount > _buffer.Length)
             {
-                var newBuffer = new Span<byte>(new byte[(Length + amount) << 1]);
-                _buffer.CopyTo(newBuffer);
-                _buffer = newBuffer;
+                if (_allowBufferResizing)
+                {
+                    var newBuffer = new Span<byte>(new byte[(Length + amount) << 1]);
+                    _buffer.CopyTo(newBuffer);
+                    _buffer = newBuffer;
+                }
+                else
+                {
+                    throw new BebopViewException("This Bebop View cannot grow the buffer.");
```

Comment:
Throwing here is of course the the easiest method to fault writing, but I think in a future change I will look into how we can propagate a `-1` up the call stack to allow an opportunity for the calling application to allocate a properly sized buffer without throwing in a potential hot path.